src/models/popUp.js

```diff
@@ -35,131 +36,17 @@ const popUpModel = {
-                let ret;
-                if (totalOpacity === 1) {  // seems to work even though a float
-                    ret = ZEROISH;
-                } else {
-                    ret = trapezoid(totalOpacity, .4, .6);
-                }
-                return ret ** coeffNearlyOpaqueOverlay;
-            }           
-
-            function monochromeOverlay(fnode) {
-                const rgba = rgbaFromString(getComputedStyle(fnode.element).getPropertyValue('background-color'));
-                return trapezoid(1 - saturation(...rgba), .96, 1) ** coeffMonochromeOverlay;
-            }    
-
-            function suspiciousClassOrIdOverlay(fnode) {
-                const element = fnode.element;
-                const attributeNames = ['class', 'id'];
-                let numOccurences = 0;
-                function numberOfSuspiciousSubstrings(value) {
-                    // return value.includes('popup') + value.includes('modal');
-                    return value.includes('popup') + value.includes('modal') + value.includes('overlay') + value.includes('underlay') + value.includes('backdrop');
-                }
-
-                for (const name of attributeNames) {
-                    let values = element.getAttribute(name);
-                    if (values) {
-                        if (!Array.isArray(values)) {
-                            values = [values];
-                        }
-                        for (const value of values) {
-                            numOccurences += numberOfSuspiciousSubstrings(value);
-                        }
-                    }
-                }
-
-                // 1 occurrence gets us to about 70% certainty; 2, 90%. It bottoms
-                // out at ZEROISH and tops out at ONEISH.
-                // TODO: Figure out how to derive the magic number .1685 from
-                // ZEROISH and ONEISH.
-                return (-((.3 + ZEROISH) ** (numOccurences + .1685)) + ONEISH) ** coeffClassOrIdOverlay;
-            }       
-            function visibleOverlay(fnode) {
-                const element = fnode.element;
-                for (const ancestor of ancestors(element)) {
-                    const style = getComputedStyle(ancestor);
-                    if (style.getPropertyValue('visibility') === 'hidden' ||
-                        style.getPropertyValue('display') === 'none') {
-                        return ZEROISH ** coeffVisibleOverlay;
-                    }
-                    // Could add opacity and size checks here, but the
-                    // "nearlyOpaque" and "big" rules already deal with opacity
-                    // and size. If they don't do their jobs, maybe repeat
-                    // their work here (so it gets a different coefficient).
-                }
-                return ONEISH ** coeffVisibleOverlay;
-            }
+            function bigOverlay(fnode) {return overlay.bigGenerator(fnode) ** coeffBigOverlay;}
```

Comment:
Good job factoring out the repetition. You can even do `score(fnode => overlay.bigGenerator(fnode) ** coeffBigOverlay)` within the ruleset itself if you want even more brevity.